src/Features/CSharp/Portable/CodeRefactorings/ConvertLocalFunctionToMethod/CSharpConvertLocalFunctionToMethodCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,230 @@
+                .WithBody(localFunction.Body);
+
+            var generator = CSharpSyntaxGenerator.Instance;
+            var editor = new SyntaxEditor(root, generator);
+            editor.InsertAfter(container, method);
+            editor.RemoveNode(localFunction, SyntaxRemoveOptions.KeepNoTrivia);
+
+            var supportsNonTrailing = SupportsNonTrailingNamedArguments(root.SyntaxTree.Options);
+
+            var enclosingTypeParameters = typeParameters.IsEmpty() ? null :
+                typeParameters
+                    .Except(declaredSymbol.TypeParameters)
+                    .Select(p => (TypeSyntax)p.Name.ToIdentifierName())
+                    .ToList();
+
+            // Update callers' arguments and type arguments
+            foreach (var currentInvocation in parentBlock.DescendantNodes().OfType<InvocationExpressionSyntax>())
+            {
+                // Using symbol to get type arguments, since it could be inferred and not present in the source
+                var symbol = semanticModel.GetSymbolInfo(currentInvocation.Expression, cancellationToken).Symbol as IMethodSymbol;
+                if (symbol?.OriginalDefinition != declaredSymbol)
+                {
+                    continue;
+                }
+
+                var shouldAddNamedArguments =
+                    !supportsNonTrailing && currentInvocation.ArgumentList.Arguments.Any(arg => arg.NameColon != null);
+
+                var capturesAsArguments = capturesAsParameters.Select(parameter => (ArgumentSyntax)generator.Argument(
+                    name: shouldAddNamedArguments ? parameter.Name : default,
+                    refKind: parameter.RefKind,
+                    expression: parameter.Name.ToIdentifierName()));
+
+                var updatedInvocation = currentInvocation.AddArgumentListArguments(capturesAsArguments.ToArray());
+
+                if (enclosingTypeParameters != null)
+                {
+                    var typeArguments = enclosingTypeParameters.Concat(symbol.TypeArguments.Select(x => x.GenerateTypeSyntax()));
+                    updatedInvocation = updatedInvocation.WithExpression(
+                            (ExpressionSyntax)generator.WithTypeArguments(updatedInvocation.Expression, typeArguments));
+                }
+
+                editor.ReplaceNode(currentInvocation, updatedInvocation);
+            }
+
+            // Update names only if needed
+            if (methodName != declaredSymbol.Name)
+            {
+                var nameToken = methodName.ToIdentifierToken();
+                foreach (var id in parentBlock.DescendantNodes().OfType<IdentifierNameSyntax>())
+                {
+                    var symbol = semanticModel.GetSymbolInfo(id, cancellationToken).Symbol as IMethodSymbol;
+                    if (symbol?.OriginalDefinition == declaredSymbol)
+                    {
+                        editor.ReplaceNode(id, WithName(id, nameToken));
```

Comment:
just to check.  do you have a test that has both a rename, and which adds type-argumetns and argument-names?  I want to make sure all the ReplaceNode calls work in that case.

Reply:
Done